File: prob3.cpp
#include <iostream>
using namespace std;

bool isFactor(long long int num, int x);
bool isPrime(int x);

int main(){

long long int thisNum = 600851475143;
int largestPfact = 0;

	for (long long int i = thisNum/2; i > 1; i = i-2)
	{
		if ((isFactor(thisNum, i)) && (isPrime(i))){
			largestPfact = i;
			break;
		}
	}
	
	cout << largestPfact << endl;

	return 0;
}

bool isFactor(long long int num, int x){
	if (num % x == 0) return true;
	else return false;
}

bool isPrime(int x){

	if (x < 2) return false;

	for (int i = 2; i < x/2; ++i)
	{
		if (isFactor(x, i)) return false;
	}

	return true;
}

File: prob3.py
#!
#incredibly slow...

import time
start = time.time()

def isFactor(num, fact):
	if num % fact == 0:
		return True
	else :
		return False

def isPrime(fact):

	for x in xrange(2, fact/2):
		if (isFactor(fact, x)):
			return False

	return True

thisNum = 600851475143
largestPfact = 0


for x in xrange(thisNum/2, thisNum/2 - 1000, -2):
	print x
	if isFactor(thisNum, x) and isPrime(x):
		largestPfact = x
		found = True
		break


print largestPfact
print 'It took', time.time()-start, 'seconds.'
